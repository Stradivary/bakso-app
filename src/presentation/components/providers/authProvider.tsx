import { Database } from "@/data/models/supabase.types";
import { signInUser } from "@/domain/services/authService";
import { supabase } from "@/domain/services/supabaseService";
import { notifications } from "@mantine/notifications";
import { Session } from "@supabase/supabase-js";
import {
  createContext,
  PropsWithChildren,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from "react";

type UserDetails = Database["public"]["Tables"]["user_profiles"]["Row"];

export interface AuthContextType {
  session: Session | null;
  user: UserDetails | null;
  isLoading: boolean;
  error: string | null;
  login: (
    name: string,
    role: string,
    latitude: number,
    longitude: number,
  ) => Promise<void>;
  logout: () => Promise<void>;
  isAuthenticated: boolean;
}

export const AuthProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const [session, setSession] = useState<Session | null>(null);
  const [user, setUser] = useState<UserDetails | null>(null);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [initialized, setInitialized] = useState(false);

  // Fetch user profile data
  const fetchUserProfile = useCallback(async (userId: string) => {
    try {
      const { data, error: fetchError } = await supabase
        .from("user_profiles")
        .select("*")
        .eq("id", userId)
        .single();

      if (fetchError) throw fetchError;

      if (data) {
        setUser(data);
      }
    } catch (error) {
      console.error("Error fetching user profile:", error);
      setError("Failed to fetch user profile");
    }
  }, []);

  // Initialize auth state
  useEffect(() => {
    const initializeAuth = async () => {
      try {
        // Check for existing session
        const {
          data: { session: existingSession },
          error: sessionError,
        } = await supabase.auth.getSession();

        if (sessionError) {
          console.error("failed to set session, log in again");
          throw sessionError;
        }

        if (existingSession) {
          setSession(existingSession);
          await fetchUserProfile(existingSession.user.id);
        }
      } catch (error) {
        console.error("Error initializing auth:", error);
        setError("Failed to initialize authentication");
      } finally {
        setIsLoading(false);
        setInitialized(true);
      }
    };

    initializeAuth();
  }, [fetchUserProfile]);
  const login = useCallback(
    async (name: string, role: string, latitude: number, longitude: number) => {
     
      try {
        setIsLoading(true);
        const { session: newSession, error: signInError } = await signInUser(
          name,
          role,
          latitude,
          longitude,
        );
        if (signInError) throw signInError;

        setSession(newSession);
        if (newSession) {
          await fetchUserProfile(newSession.user.id);
        }
        setError(null);
      } catch (error) {
        console.error("Error during login:", error);
        setError("Failed to login");
      } finally {
        setIsLoading(false);
      }
    },
    [fetchUserProfile],
  );

  const logout = useCallback(async () => {
    try {
      const { error: signOutError } = await supabase.auth.signOut({
        scope: "local",
      });

      if (signOutError) throw signOutError;
      setSession(null);
      setUser(null);
    } catch (error) {
      console.error("Error during logout:", error);
      setError("Failed to logout");
    }
  }, []);

  // Storage change observer
  useEffect(() => {
    if (!initialized) return;

    const handleStorageChange = (event: StorageEvent) => {
      const currentSession = event.newValue;
      const previousSession = event.oldValue;

      if (event.key === "abangbakso-session") {
        if (!previousSession) {
          notifications.show({
            title: "Warning",
            message: "Possible session injection detected",
            color: "orange",
          });
          logout();
          setError("Session security violation detected");
          return;
        }

        if (!currentSession) {
          notifications.show({
            title: "Warning",
            message: "Possible session removal detected",
            color: "orange",
          });
          logout();
          setError("Session security violation detected");
          return;
        }
        // If session was changed externally
        if (currentSession !== previousSession) {
          notifications.show({
            title: "Warning",
            message: "Possible session tampering detected",
            color: "orange",
          });
          logout();
          setError("Session security violation detected");
        }
      }
    };

    // Add storage event listener
    window.addEventListener("storage", handleStorageChange);

    return () => {
      window.removeEventListener("storage", handleStorageChange);
    };
  }, [initialized, logout, session?.user.id]);

  useEffect(() => {
    if (!initialized) return;

    const {
      data: { subscription },
    } = supabase.auth.onAuthStateChange(async (event, newSession) => { 

      if (!newSession?.user.id) return;
      if ((newSession?.user.id && newSession?.user.id) && newSession?.user.id !== session?.user.id) {
        return;
      }
      switch (event) {
        case "INITIAL_SESSION":
          break;
        case "SIGNED_IN":
        case "TOKEN_REFRESHED":
        case "USER_UPDATED":
          setSession(newSession);
          if (newSession) {
            await fetchUserProfile(newSession.user.id);
          }
          break;
        case "SIGNED_OUT":
          setSession(null);
          setUser(null);
          break;
      }
    });

    return () => {
      subscription.unsubscribe();
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [initialized, fetchUserProfile]);

  const isAuthenticated = !!session?.user?.id && !isLoading;

  const values = useMemo(
    () => ({
      session,
      user,
      isLoading,
      error,
      login,
      logout,
      isAuthenticated,
    }),
    [session, user, isLoading, error, login, logout, isAuthenticated],
  );

  return <AuthContext.Provider value={values}>{children}</AuthContext.Provider>;
};

export const AuthContext = createContext<AuthContextType | undefined>(
  undefined,
);

export const useAuth = () => {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error("useAuth must be used within an AuthProvider");
  }
  return context;
};
