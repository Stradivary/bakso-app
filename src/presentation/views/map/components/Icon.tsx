import L from "leaflet";
import markerPerson from "@/shared/assets/cust.svg";
import markerBakso from "@/shared/assets/seller.svg";

const iconPerson = new L.Icon({
  iconUrl: markerPerson,
  iconSize: new L.Point(32, 49),
});

const iconBakso = new L.Icon({
  iconUrl: markerBakso,
  iconSize: new L.Point(32, 49),
});

export { iconPerson, iconBakso };
