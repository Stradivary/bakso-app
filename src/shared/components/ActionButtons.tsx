import { Badge, Box, Button, Flex, Group, Stack } from "@mantine/core";
import { LocateFixed, X } from "lucide-react";
import { NotificationCenter } from "./NotificationCenter";
import { ActionButtonProps } from "../models/actionButtons.types";

export const ActionButtons = ({
  onExit,
  onRecenter,
  role,
  name,
  notifications,
}: ActionButtonProps) => (
  <Box
    style={{
      zIndex: 50,
      position: "absolute",
      top: 0,
      right: 0,
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-end",
      padding: 8,
    }}
  >
    <Flex direction="row">
      <Group justify="end" align="start" m={8} gap={6} style={{
        userSelect: 'none',
        pointerEvents: 'painted'
      }}>
        <Badge color={role === "seller" ? "green" : "blue"}>
          {role === "seller" ? "Tukang Bakso" : "Customer"}
        </Badge>
        <Badge variant="filled" color="teal">
          {name}
        </Badge>
      </Group>
      <Stack m={8} style={{ flexGrow: 1 }} align="end">
        <Button onClick={onRecenter} variant="white" size="compact-lg" w={50}>
          <LocateFixed size={20} />
        </Button>
        <Button
          color="red"
          onClick={onExit}
          variant="white"
          size="compact-lg"
          w={50}
        >
          <X size={20} />
        </Button>
        {role === "seller" && (
          <NotificationCenter notifications={notifications} />
        )}
      </Stack>

    </Flex>
  </Box>
);
