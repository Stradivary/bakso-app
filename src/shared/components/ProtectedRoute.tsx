import { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { useAuth } from "../hooks/useAuth";

export const ProtectedRoute: React.FC<{ children: JSX.Element; }> = ({
  children,
}) => {
  const { isAuthenticated } = useAuth();
  const navigate = useNavigate();
  useEffect(() => {
    if (!isAuthenticated) {

      navigate('/login')
    }
  }, [isAuthenticated, navigate])
  return children;
};
