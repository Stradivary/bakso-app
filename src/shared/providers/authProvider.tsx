import { Session } from '@supabase/supabase-js';
import { PropsWithChildren, useCallback, useEffect, useState, useRef } from 'react';
import { Database } from '../models/supabase';
import { signInUser } from '../services/authService';
import { supabase } from '../services/supabaseService';
import { AuthContext } from '../hooks/useAuth';

type UserDetails = Database['public']['Tables']['user_profiles']['Row'];

export interface AuthContextType {
  session: Session | null;
  userDetails: UserDetails | null;
  isLoading: boolean;
  error: string | null;
  login: (name: string, role: string, latitude: number, longitude: number) => Promise<void>;
  logout: () => Promise<void>;
  isAuthenticated: boolean;
}

export const AuthProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const [session, setSession] = useState<Session | null>(null);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  
  const validSessionRef = useRef<string | null>(null);
  
  const validateSessionIntegrity = useCallback(async () => {
    try {

      const { data: { session: currentSession }, error } = await supabase.auth.getSession();
      
      if (error) throw error;

      const storedSession = window.sessionStorage.getItem('supabase.auth.token');
      
      if (storedSession && validSessionRef.current && storedSession !== validSessionRef.current) {
        console.error('Session storage tampering detected');
        await logout();
        setError('Security violation detected. Please login again.');
        return false;
      }
      
      if (currentSession?.access_token) {
        validSessionRef.current = currentSession.access_token;
      }
      
      return true;
    } catch (error) {
      console.error('Error validating session integrity:', error);
      return false;
    }
  }, []);

  const login = async (name: string, role: string, latitude: number, longitude: number) => {
    setIsLoading(true);
    try {
      const { session, error } = await signInUser(name, role, latitude, longitude);
      if (error) throw error;
      
      if (session?.access_token) {
        validSessionRef.current = session.access_token;
      }
      
      setSession(session);
      setError(null);
    } catch (error) {
      console.error('Error during login:', error);
      setError('Failed to login');
    } finally {
      setIsLoading(false);
    }
  };

  const logout = async () => {
    try {
      await supabase.auth.signOut();
      setSession(null);
      setError(null);
      validSessionRef.current = null;
      
      window.sessionStorage.removeItem('supabase.auth.token');
    } catch (error) {
      console.error('Error during logout:', error);
      setError('Failed to logout');
    }
  };

  const handleAuthStateChange = useCallback(async (_event: string, newSession: Session | null) => {
    const isValid = await validateSessionIntegrity();
    if (isValid) {
      setSession(newSession);
    } else {
      await logout();
    }
  }, [validateSessionIntegrity]);

  useEffect(() => {
    const initializeAuth = async () => {
      setIsLoading(true);
      try {
        const { data: { session: initialSession }, error } = await supabase.auth.getSession();
        if (error) {
          console.error('Error getting initial session:', error);
          setError('Failed to get initial session');
        } else {
          if (initialSession?.access_token) {
            validSessionRef.current = initialSession.access_token;
          }
          setSession(initialSession);
        }
      } catch (error) {
        console.error('Error getting initial session:', error);
        setError('Failed to get initial session');
      } finally {
        setIsLoading(false);
      }
    };

    initializeAuth();

    const handleStorageChange = async (event: StorageEvent) => {
      if (event.key && event.key.includes('supabase.auth')) {
        const isValid = await validateSessionIntegrity();
        if (!isValid) {
          await logout();
        }
      }
    };

    window.addEventListener('storage', handleStorageChange);

    const validationInterval = setInterval(validateSessionIntegrity, 30000); // Check every 30 seconds

    const { data: { subscription } } = supabase.auth.onAuthStateChange(handleAuthStateChange);

    return () => {
      window.removeEventListener('storage', handleStorageChange);
      clearInterval(validationInterval);
      subscription.unsubscribe();
    };
  }, [handleAuthStateChange, validateSessionIntegrity]);

  const isAuthenticated = !!session;

  return (
    <AuthContext.Provider value={{
      session,
      userDetails: null,
      isLoading,
      error,
      login,
      logout,
      isAuthenticated
    }}>
      {children}
    </AuthContext.Provider>
  );
};