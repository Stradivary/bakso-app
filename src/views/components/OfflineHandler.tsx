import { Modal, Title } from "@mantine/core";
import { useNetwork } from "@mantine/hooks";

export function OfflineHandler() {
  const { online } = useNetwork();
  return (
    <Modal
      title={
        <Title order={5} style={{ margin: 0 }}>
          Duh, Koneksi Internet Terputus!
        </Title>
      }
      opened={!online}
      onClose={() => {}}
      withCloseButton={false}
    >
      Sepertinya koneksi internet Anda terputus. Mohon periksa kembali koneksi
      internet Anda.
    </Modal>
  );
}
