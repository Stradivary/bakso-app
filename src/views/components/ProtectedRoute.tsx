import { Box, LoadingOverlay } from "@mantine/core";
import { Navigate } from "react-router-dom";
import { useAuth } from "@/shared/contexts/authProvider";

export const ProtectedRoute: React.FC<{ children: JSX.Element }> = ({
  children,
}) => {
  const { isAuthenticated, isLoading } = useAuth();

  if (isLoading) {
    return (
      <Box h="100dvh">
        <LoadingOverlay visible />
      </Box>
    );
  }

  if (!isAuthenticated) {
    return <Navigate to="/login" replace />;
  }

  return children;
};
