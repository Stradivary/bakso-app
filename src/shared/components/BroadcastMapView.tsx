import { Badge, Button, Center, Image, Stack, Text } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { modals } from '@mantine/modals';
import L, { LatLng, Point } from 'leaflet';
import React, { useEffect } from 'react';
import { MapContainer, Marker, Popup, TileLayer, Tooltip } from 'react-leaflet';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../hooks/useAuth';
import { useLocation } from '../hooks/useLocation';
import { calculateDistance, useLocationTracking } from '../hooks/useTracker';
import { ActionButtons } from './ActionButtons';
import { Dialog } from './Dialog';
import { iconBakso, iconPerson } from "./Icon";

const UPDATE_INTERVAL = 5 * 1000; // Update interval in milliseconds (5 seconds)

// Custom hook to handle map updates
const MapUpdater = ({
  userRole,
  updateLocation,
}: {
  userRole: string,
  updateLocation: (location: L.LatLng) => void;
}
) => {
  const { location } = useLocation();

  useEffect(() => {
    if (userRole === 'seller') {
      const updateInterval = setInterval(() => {
        updateLocation(new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0));
      }, UPDATE_INTERVAL);

      return () => clearInterval(updateInterval);
    }
  }, [userRole, updateLocation, location?.latitude, location?.longitude]);

  return null;
};

const BroadcastMapView: React.FC = () => {
  const navigate = useNavigate();
  const { session, logout } = useAuth();

  const userId = session?.user?.id as string;
  const userRole = sessionStorage.getItem('role') as 'seller' | 'buyer';
  const [exitModalOpened, { close: exitModalClose, open: openModal }] = useDisclosure();

  const { location } = useLocation();


  const {
    nearbyUsers,
    updateLocation,
    pingSeller,
    deactivateUser,
    notifications
  } = useLocationTracking(userId, userRole, new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0), session?.user?.user_metadata?.name ?? '');


  const [mapRef, setMapRef] = React.useState<L.Map | null>(null);

  const handleRecenter = React.useCallback(() => {
    if (mapRef && location) {
      mapRef.setView(new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0), mapRef.getZoom(), {
        animate: true,
        duration: 1,
      });
    }
  }, [mapRef, location]);

  const handleExit = React.useCallback(() => {
    deactivateUser();
    logout();
    sessionStorage.removeItem('role');
    exitModalClose();
  }, [deactivateUser, exitModalClose, logout, navigate]);

  if (!location) {
    return (
      <div>
        <Text>Getting your location...</Text>
      </div>
    );
  }

  return (
    <>
      <MapContainer
        key={location.toString()}
        ref={(map) => setMapRef(map)}
        center={new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0) ?? new LatLng(0, 0)}
        zoom={16}
        minZoom={13}
        maxZoom={18}
        style={{ height: "100dvh", width: '100%', zIndex: 0 }}
      >
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />


        {userRole === 'seller' && (
          <MapUpdater userRole={userRole} updateLocation={updateLocation} />
        )}

        {/* Render user's current location */}
        <Marker position={new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0)} icon={userRole === 'seller' ? iconBakso : iconPerson}>
          <Tooltip permanent direction='bottom' offset={new Point(0, 12)}>
            Lokasi
            Anda
          </Tooltip>
        </Marker>

        {/* Render nearby users */}
        {nearbyUsers.map((nearbyUser) => {
          return (
            <Marker
              key={`${nearbyUser.user_id} ${nearbyUser.role}`}
              position={new L.LatLng(nearbyUser?.location?.lat ?? 0, nearbyUser?.location?.lng ?? 0)}
              icon={nearbyUser.role === 'seller' ? iconBakso : iconPerson}
              eventHandlers={{
                click: () => {
                  if (nearbyUser.role === "seller") {
                    const buyerLocation = new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0);
                    const sellerLocation = new L.LatLng(
                      nearbyUser.location.lat,
                      nearbyUser.location.lng
                    );
                    const distanceInMeters = calculateDistance(buyerLocation, sellerLocation);
                    const walkingSpeed = 1.4; // meters per second
                    const estimatedTimeInMinutes = Math.round((distanceInMeters / walkingSpeed) / 60);

                    modals.openConfirmModal({
                      title: "Pesan Bakso",
                      children: (
                        <>
                          <Text size="sm">
                            üçú Apakah Anda ingin memesan bakso dari {nearbyUser.userName}?
                          </Text>
                          <Text size="xs" c="dimmed">
                            Estimasi waktu: {estimatedTimeInMinutes} menit berjalan kaki
                          </Text>
                        </>
                      ),
                      labels: { cancel: "Batal", confirm: "Pesan" },
                      onConfirm: () => {
                        pingSeller?.(nearbyUser.user_id);
                        modals.open({
                          title: "Pesan Bakso",
                          children: "Pesan bakso telah dikirim",
                          color: "blue",
                        });
                      },
                    });
                  }
                },
              }}
            >
              <Popup>
                <div   >
                  <Text>{nearbyUser.userName}</Text>
                  <Text c="gray">{nearbyUser.role === 'seller' ? 'Pedagang Bakso' : 'Pelanggan'}</Text>

                </div>
              </Popup>
            </Marker>
          );
        })}
      </MapContainer>

      <ActionButtons
        role={userRole}
        isLoading={false}
        onExit={() => {
          openModal();
        }}
        notifications={notifications}
        onRecenter={handleRecenter}
      >
        <Badge variant='filled' color='teal'>
          {session?.user?.user_metadata?.name}
        </Badge>
      </ActionButtons>

      <Dialog opened={exitModalOpened} close={exitModalClose}>
        <Stack>
          <Center>
            <Image src="/confirmation.png" alt="confirmation" w={100} h={100} />
          </Center>
          <Text ta="center">
            Dengan menutup halaman ini, kamu akan keluar dari pantauan Tukang
            Bakso
          </Text>
          <Button onClick={handleExit} radius="lg" size="md">
            OK
          </Button>
          <Button onClick={exitModalClose} radius="lg" variant="outline">
            Batal
          </Button>
        </Stack>
      </Dialog>
    </>
  );
};

export default BroadcastMapView;