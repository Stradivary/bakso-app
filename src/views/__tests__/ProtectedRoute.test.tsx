import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { describe, it, expect, vi, beforeEach } from "vitest";
import { ProtectedRoute } from "../components/ProtectedRoute";
import { PropsWithChildren } from "react";
import { useAuth } from "@/shared/contexts/authProvider";

// Mock the useAuth hook
vi.mock("@/shared/contexts/authProvider");

// Mock Mantine components
vi.mock("@mantine/core", () => ({
  Box: ({ children, ...props }: PropsWithChildren) => (
    <div data-testid="mantine-box" {...props}>
      {children}
    </div>
  ),
  LoadingOverlay: ({ visible, ...props }: { visible: boolean }) => (
    <div
      data-testid="mantine-loading-overlay"
      aria-hidden={!visible}
      {...props}
    />
  ),
}));

describe("ProtectedRoute", () => {
  // Helper function to setup the component with mocked routes
  const renderProtectedRoute = (
    authState: Partial<ReturnType<typeof useAuth>>,
  ) => {
    vi.mocked(useAuth).mockReturnValue({
      isAuthenticated: false,
      isLoading: false,
      ...authState,
    } as ReturnType<typeof useAuth>);

    return render(
      <MemoryRouter>
        <ProtectedRoute>
          <div data-testid="protected-content">Protected Content</div>
        </ProtectedRoute>
      </MemoryRouter>,
    );
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("shows loading overlay when authentication is being checked", () => {
    renderProtectedRoute({ isLoading: true });

    expect(screen.getByTestId("mantine-box")).toMatchSnapshot();
    // @ts-expect-error extended by testing-library
    expect(screen.getByTestId("mantine-loading-overlay")).toHaveAttribute(
      "aria-hidden",
      "false",
    );
  });

  it("redirects to login when user is not authenticated", () => {
    const { container } = renderProtectedRoute({ isAuthenticated: false });

    expect(container).toMatchSnapshot();
    // The Navigate component will be rendered with 'to="/login"'
    expect(container.innerHTML).toBe("");
  });

  it("renders protected content when user is authenticated", () => {
    renderProtectedRoute({ isAuthenticated: true });

    expect(screen.getByTestId("protected-content")).toMatchSnapshot();
    // @ts-expect-error extended by testing-library
    expect(screen.getByTestId("protected-content")).toBeInTheDocument();
    // @ts-expect-error extended by testing-library
    expect(screen.getByText("Protected Content")).toBeInTheDocument();
  });
});
