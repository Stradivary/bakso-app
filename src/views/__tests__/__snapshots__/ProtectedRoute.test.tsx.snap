// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProtectedRoute > redirects to login when user is not authenticated 1`] = `<div />`;

exports[`ProtectedRoute > renders protected content when user is authenticated 1`] = `
<div
  data-testid="protected-content"
>
  Protected Content
</div>
`;

exports[`ProtectedRoute > shows loading overlay when authentication is being checked 1`] = `
<div
  data-testid="mantine-box"
  h="100dvh"
>
  <div
    aria-hidden="false"
    data-testid="mantine-loading-overlay"
  />
</div>
`;
